lib/c-api/src/wasm_c_api/externals/function.rs

```diff
@@ -108,13 +108,15 @@ pub unsafe extern "C" fn wasm_func_new_with_env(
     let func_sig = &function_type.inner().function_type;
     let num_rets = func_sig.results().len();
 
-    #[derive(wasmer::WasmerEnv, Clone)]
+    #[derive(Clone)]
     #[repr(C)]
     struct WrapperEnv {
         env: *mut c_void,
         env_finalizer: Arc<Option<wasm_env_finalizer_t>>,
     }
 
+    impl wasmer_api::WasmerEnv for WrapperEnv {}
+
```

Comment:
Ummm.... of everything this is the thing that I think could be more inconsistent. I'm concerned it might brake some things as `derive(wasmer::WasmerEnv)` has more implications in general than just the `impl`.

For `wasmer_js` we also have this problem (where WasmerEnv is defined in `wasmer_js`). Perhaps we can have a wasmer_derive feature flag to use one or other master crate name?

Also, since `wasmer_api::WasmerEnv` works... why it wouldn't work this?
```rust
    #[derive(wasmer_api::WasmerEnv, Clone)]
    #[repr(C)]
    struct WrapperEnv {
        env: *mut c_void,
        env_finalizer: Arc<Option<wasm_env_finalizer_t>>,
    }
```